src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems

```diff
@@ -426,6 +422,11 @@
     <Compile Include="$(MSBuildThisFileDirectory)System\IO\UnmanagedMemoryAccessor.cs" />
     <Compile Include="$(MSBuildThisFileDirectory)System\IO\UnmanagedMemoryStream.cs" />
     <Compile Include="$(MSBuildThisFileDirectory)System\IO\UnmanagedMemoryStreamWrapper.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)System\IO\Strategies\BufferedFileStreamStrategy.cs" />
```

Comment:
I was thinking on a FileStream folder rather than a Strategies folder, containing also `FileStream.cs` itself.

Reply:
> on a FileStream folder

The problem with folders is that the folder name needs to match the namespace and the namespace can not be equal to the type name (compiler error). So I can't call it `File` or `FileStream`.

Reply:
Along with a preference for nested types, I would use a dotted naming convention rather than a folder,  e.g. FileStream.BufferedStrategy.cs. we can also remove the extra ceremony from the type names when everything is nested.